lib/grant-types/password-grant-type.js

```diff
@@ -56,7 +56,7 @@ PasswordGrantType.prototype.handle = function(request, client) {
     throw new InvalidArgumentError('Missing parameter: `client`');
   }
 
-  var scope = this.getScope(request);
+  const scope = this.getScope(request);
```

Comment:
This should be `let scope = ...`

Reply:
I just checked and if we return something from a function (which happens with scope) it gets automatically re-assignable:

```js
const getx = () => {
  let x = 0
  return x
}

const gety = () => {
  const y = 0
  return y
}

let a = getx()
a = 2 // works

let b = gety()
b = 2 // works, too
```

This also works with objects and object properties are assignable even with const objects (as long they were not freezed)

Reply:
Oh wow, I was under the impression that const were immutable (not object properties) even outside of the scope.  Ok, looks good then.

Reply:
Here's a good article if you're interested: https://mathiasbynens.be/notes/es6-const